lib/puppet_x/coi/jboss/provider/abstract_jboss_cli.rb

```diff
@@ -5,52 +5,58 @@
 # Base class for all JBoss providers
 class Puppet_X::Coi::Jboss::Provider::AbstractJbossCli < Puppet::Provider
 
-  # attr_writer :compilator
-
+  # Default constructor that will also initialize 3 external object, system_executor, compilator and command executor
   def initialize(resource=nil)
     super(resource)
-    @system_executor = Puppet_X::Coi::Jboss::Internal::JbossSystemExec.new
+    system_executor = Puppet_X::Coi::Jboss::Internal::JbossSystemExec.new
     @compilator = Puppet_X::Coi::Jboss::Internal::JbossCompilator.new
+    @runner = Puppet_X::Coi::Jboss::Internal::JbossRunner.new(system_executor)
+  end
+
+  def runner=(value)
+    @runner.system_executor = value
   end
 
   @@bin = "bin/jboss-cli.sh"
   @@contents = nil
-  # @@system_executor = Puppet_X::Coi::Jboss::Internal::JbossSystemExec.new
-  # @@compilator = Puppet_X::Coi::Jboss::Internal::JbossCompilator.new
-
-  class << self
-    def jbossclibin
-      home = self.jbosshome
-      path = "#{home}/#{@@bin}"
-      return path
-    end
 
-    def jbosshome
-      Puppet_X::Coi::Jboss::Configuration::config_value :home
-    end
+  # Method that returns jboss-cli command path
+  # @return {String} jboss-cli command path
+  def jbossclibin
+    home = self.jbosshome
+    path = "#{home}/#{@@bin}"
+    path
+  end
 
-    def jbosslog
-      Puppet_X::Coi::Jboss::Configuration::config_value :console_log
-    end
+  # CONFIGURATION VALUES
 
-    def config_runasdomain
-      Puppet_X::Coi::Jboss::Configuration::config_value :runasdomain
-    end
+  # Method that returns jboss home value
+  # @return {String} home value
+  def jbosshome
+    Puppet_X::Coi::Jboss::Configuration::config_value :home
+  end
 
-    def config_controller
-      Puppet_X::Coi::Jboss::Configuration::config_value :controller
-    end
+  def jbosslog
+    Puppet_X::Coi::Jboss::Configuration::config_value :console_log
+  end
 
-    def config_profile
-      Puppet_X::Coi::Jboss::Configuration::config_value :profile
-    end
+  def config_runasdomain
+    Puppet_X::Coi::Jboss::Configuration::config_value :runasdomain
+  end
+
+  def config_controller
+    Puppet_X::Coi::Jboss::Configuration::config_value :controller
+  end
 
+  def config_profile
+    Puppet_X::Coi::Jboss::Configuration::config_value :profile
   end
 
+
   # TODO: Uncomment for defered provider confinment after droping support for Puppet < 3.0
   # commands :jbosscli => Puppet_X::Coi::Jboss::Provider::AbstractJbossCli.jbossclibin
 
-  def runasdomain?
```

Comment:
!fix Revert this to original or explain this change. `is_runasdomain` is a bit Java convention. Ruby conventions for boolean returning method are to ends them with `?` sign without `is`

Here: http://stackoverflow.com/a/1345854/844449 and here https://github.com/bbatsov/ruby-style-guide#bool-methods-qmark also here https://github.com/bbatsov/rubocop/blob/master/config/enabled.yml#L554-L557